algoliasearch-common/src/main/java/com/algolia/search/APIClient.java

```diff
@@ -807,6 +807,12 @@ private TaskSingleIndex batchSingleIndex(
 
   Task deleteObject(String indexName, String objectID, RequestOptions requestOptions)
       throws AlgoliaException {
+
+    if (objectID.trim().length() == 0)
+    {
+      throw new AlgoliaException("ObjectID must not be empty");
```

Comment:
nit: What do you think of throwing an instance of a subclass of `AlgoliaException`? Same reason as usual, from the user POV it may be easier to catch this specific error.

Reply:
Could we use this built-in Java exception ?
https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html

Reply:
build in java is nice 👍

Reply:
👍